src/btree/bt_split.c

```diff
@@ -1775,8 +1776,9 @@ __split_insert(WT_SESSION_IMPL *session, WT_REF *ref)
      * otherwise the page might be evicted based on its last reconciliation which no longer matches
      * reality after the split.
      */
-    WT_ASSERT(session, __wt_leaf_page_can_split(session, page));
-    WT_ASSERT(session, __wt_page_is_modified(page));
+    WT_ASSERT_OPTIONAL(session, __wt_leaf_page_can_split(session, page),
+      "Attempting to split page that does not require splitting");
```

Comment:
"Attempting to split page that does not require splitting." I don't believe that is exactly what is wrong here. A more accurate message would be along the lines of: "Attempted to split a page that cannot be split."